common/src/test/java/com/microsoft/identity/common/SharedPreferencesAccountCredentialCacheTest.java

```diff
@@ -1401,6 +1401,127 @@ public void getCorrectCredentialWhenRequestedClaimsAreSpecified() {
         assertEquals("SecretB", credentials.get(0).getSecret());
     }
 
+    @Test
+    public void getExactMatchingCredentialWhenMustMatchExactClaimsAreSpecified() {
+        final RefreshTokenRecord refreshToken = new RefreshTokenRecord();
+        refreshToken.setSecret(SECRET);
+        refreshToken.setHomeAccountId(HOME_ACCOUNT_ID);
+        refreshToken.setEnvironment(ENVIRONMENT);
+        refreshToken.setCredentialType(CredentialType.RefreshToken.name());
+        refreshToken.setClientId(CLIENT_ID);
+        refreshToken.setTarget(TARGET);
+
+        final AccessTokenRecord accessToken = new AccessTokenRecord();
+        accessToken.setCachedAt(CACHED_AT);
+        accessToken.setExpiresOn(EXPIRES_ON);
+        accessToken.setSecret("SecretA");
+        accessToken.setHomeAccountId(HOME_ACCOUNT_ID);
+        accessToken.setRealm(REALM);
+        accessToken.setEnvironment(ENVIRONMENT);
+        accessToken.setCredentialType(CredentialType.AccessToken.name());
+        accessToken.setClientId(CLIENT_ID);
+        accessToken.setApplicationIdentifier(APPLICATION_IDENTIFIER_SHA512);
+        accessToken.setMamEnrollmentIdentifier(MAM_ENROLLMENT_IDENTIFIER);
+        accessToken.setTarget(TARGET);
+        // No requested claims specified for accessToken
+
+        final AccessTokenRecord accessToken2 = new AccessTokenRecord();
+        accessToken2.setCachedAt(CACHED_AT);
+        accessToken2.setExpiresOn(EXPIRES_ON);
+        accessToken2.setSecret("SecretB");
+        accessToken2.setHomeAccountId(HOME_ACCOUNT_ID);
+        accessToken2.setRealm(REALM);
+        accessToken2.setEnvironment(ENVIRONMENT);
+        accessToken2.setCredentialType(CredentialType.AccessToken.name());
+        accessToken2.setClientId(CLIENT_ID);
+        accessToken2.setApplicationIdentifier(APPLICATION_IDENTIFIER_SHA512);
+        accessToken2.setMamEnrollmentIdentifier(MAM_ENROLLMENT_IDENTIFIER);
+        accessToken2.setTarget(TARGET);
+        accessToken2.setRequestedClaims("{\"access_token\":{\"deviceid\":{\"essential\":true}}}");
+
+        // Save the Credentials
+        mSharedPreferencesAccountCredentialCache.saveCredential(refreshToken);
+        mSharedPreferencesAccountCredentialCache.saveCredential(accessToken);
+        mSharedPreferencesAccountCredentialCache.saveCredential(accessToken2);
+
+        List<Credential> credentials = mSharedPreferencesAccountCredentialCache.getCredentialsFilteredBy(
+                HOME_ACCOUNT_ID,
+                ENVIRONMENT,
+                CredentialType.AccessToken,
+                CLIENT_ID,
+                APPLICATION_IDENTIFIER_SHA512,
+                MAM_ENROLLMENT_IDENTIFIER,
+                REALM,
+                null,
+                BEARER_AUTHENTICATION_SCHEME.getName(),
+                null,
```

Comment:
>null [](http://example.com/codeflow?start=16&length=4)

What's expect if requestedClaims is "" instead null? Should it behave same as null? Seems currently they would behave differently.

Reply:
Thanks for pointing this out. Made changes to handle both null and empty as same and added test case for it.